### PR TITLE
Streamline cluster and vectorize

### PR DESCRIPTION
**Pull request by jtwinrelevanceai**

### CODE REVIEW

File: docsrc/manual_source/client.rst [hunk 1]
```diff
@@ -2,5 +2,11 @@ Client
 ========================================
 
 .. automodule:: relevanceai.client.client
+<<<<<<< HEAD
```
Comment (boba-and-beer) [inline docsrc/manual_source/client.rst:5]: Fix merge conflict here

File: docsrc/manual_source/relevanceai.operations.unstruc.audio.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+relevanceai.operations.unstruc.audio package
```
Comment (boba-and-beer) [inline docsrc/manual_source/relevanceai.operations.unstruc.audio.rst:1]: Remove this please unless this was manually edited

File: docsrc/manual_source/relevanceai.operations.unstruc.image.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+relevanceai.operations.unstruc.image package
```
Comment (boba-and-beer) [inline docsrc/manual_source/relevanceai.operations.unstruc.image.rst:1]: Remove this please unless this was manually edited

File: docsrc/manual_source/relevanceai.operations.unstruc.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
+relevanceai.operations.unstruc package
+======================================
+
```
Comment (boba-and-beer) [inline docsrc/manual_source/relevanceai.operations.unstruc.rst:3]: Remove this please unless this was manually edited

File: docsrc/manual_source/relevanceai.operations.unstruc.text.sentiment.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
+relevanceai.operations.unstruc.text.sentiment package
```
Comment (boba-and-beer) [inline docsrc/manual_source/relevanceai.operations.unstruc.text.sentiment.rst:1]: Remove this please unless this was manually edited

File: relevanceai/utils/progress_bar.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ def is_in_notebook(self) -> bool:
         return is_in_notebook
 
     def get_bar(self):
-        from tqdm.auto import tqdm
+        from tqdm import tqdm
```
Comment (boba-and-beer) [inline relevanceai/utils/progress_bar.py:41]: you need `auto` to provide compatibility between notebooks and scripts

File: relevanceai/utils/progress_bar.py [hunk 7]
```diff
@@ -74,7 +74,12 @@ def progress_bar(iterable, show_progress_bar: bool = False):
 
     try:
         if show_progress_bar:
-            return ProgressBar()(iterable)
+            from tqdm import tqdm
```
Comment (boba-and-beer) [inline relevanceai/utils/progress_bar.py:77]: `from tqdm.auto`

File: relevanceai/operations/vector/vectorize.py [hunk 8]
```diff
@@ -161,108 +142,286 @@ def vectorize(
...
-                        text_encoder.encode_documents(new_text_fields, documents)
-                    )
+    def _get_filters(
+        self, fields: List[str], vector_fields: List[str]
+    ) -> List[Dict[str, Any]]:
 
-                return updated_documents
+        filters = []
+        if len(fields) > 1:
+            iters = len(fields) ** 2
```
Comment (boba-and-beer) [inline relevanceai/operations/vector/vectorize.py:169]: Can you leave a comment explaining what we are doing here and why thanks

File: relevanceai/operations/vector/vectorize.py [hunk 9]
```diff
@@ -1,156 +1,137 @@
...
+                model = MobileNetV12Vec()
 
-            df.vectorize(
-                image_fields=["image_field_1", "image_field_2"],
-                text_fields=["text_field"],
-                text_model=text_model
-            )
+        else:
+            # TODO: this needs to be referenced from relevance.constants.errors
+            raise ValueError("ModelNotSupported")
```
Comment (boba-and-beer) [inline relevanceai/operations/vector/vectorize.py:78]: Model Not Supported Error

File: relevanceai/dataset/write/write.py [hunk 10]
```diff
@@ -570,7 +570,7 @@ def cat_fields(documents, field_name):
             ]
             return cat_vector_documents
 
-        self.pull_update_push(
+        self.pull_update_push_async(
```
Comment (boba-and-beer) [inline relevanceai/dataset/write/write.py:573]: odd

File: relevanceai/operations/cluster/cluster.py [hunk 11]
```diff
@@ -131,7 +129,7 @@ def _get_package(self, model):
         elif "hdbscan" in model_name:
             package = "hdbscan"
 
-        elif "community_detection" in model_name:
+        elif "communitydetection" in model_name:
```
Comment (boba-and-beer) [inline relevanceai/operations/cluster/cluster.py:132]: huh

Comment (jtwinrelevanceai) [reply relevanceai/operations/cluster/cluster.py:132]: Model name has underscores recovered at beginning of func

File: relevanceai/operations/cluster/algorithms/community_detection.py [hunk 12]
```diff
@@ -4,8 +4,16 @@
 
 
 class CommunityDetection:
-    def __init__(self, gpu=False):
+    def __init__(self, config, gpu=False):
+
         self.gpu = gpu
+        self.threshold = 0.75
```
Comment (boba-and-beer) [inline relevanceai/operations/cluster/algorithms/community_detection.py:10]: hard-coded values should be at the top - never set as an attribute later on as this should be easily adjusted.

Comment (jtwinrelevanceai) [reply relevanceai/operations/cluster/algorithms/community_detection.py:10]: true

**Review by boba-and-beer — APPROVED** [review]

File: relevanceai/operations/operations.py [hunk 13]
```diff
@@ -17,8 +17,8 @@ def __init__(
 
     def cluster(
         self,
-        model: Union[str, Any],
-        vector_fields: List[str],
+        model: Any = None,
+        vector_fields: List[str] = [],
```
Comment (boba-and-beer) [inline relevanceai/operations/operations.py:21]: this needs to be None. Do not have default list arguments.

**Review by boba-and-beer — APPROVED** [review]

**Pull request merged**